test/mobile/custom_build/predictor.cpp

```diff
@@ -0,0 +1,52 @@
+#include <iostream>
+#include <string>
+#include <torch/script.h>
+
+using namespace std;
+
+namespace {
+
+struct MobileCallGuard {
+  // AutoGrad is disabled for mobile by default.
+  torch::autograd::AutoGradMode no_autograd_guard{false};
+  // Disable graph optimizer to ensure list of unused ops are not changed for
+  // custom mobile build.
+  torch::jit::GraphOptimizerEnabledGuard no_optimizer_guard{false};
+};
+
+void init() {
+  auto qengines = at::globalContext().supportedQEngines();
+  if (std::find(qengines.begin(), qengines.end(), at::QEngine::QNNPACK) !=
+      qengines.end()) {
+    at::globalContext().setQEngine(at::QEngine::QNNPACK);
```

Comment:
Also not really related to this PR, but is there really any reason why you wouldn't want the lack of QNNPACK to be a hard error? This code (1) seems annoying boilerplate that everyone has to write, and (2) seems like it would mask build problems when QNNPACK is not available.

Reply:
Ideally, it wouldn't be necessary to write this boilerplate at all! I guess it would be better to figure out who added this in the first place and ask them about it. @jerryzh168, any comments here?

Reply:
> Ideally, it wouldn't be necessary to write this boilerplate at all! I guess it would be better to figure out who added this in the first place and ask them about it. @jerryzh168, any comments here?

I think this is introduced since @supriyar unified fbgemm & qnnpack interface at dispatcher level. Now they look the same but the underlying packed weights format / etc are different. So we introduced the concept called "qengine" to switch between these two quantization implementations. And it's possible to have both qnnpack/fbgemm available in one build so I guess it needs to be set explicitly. Yet another dispatching mechanism? :)

Reply:
For mobile build though we turn off FBGEMM, right? In that case the default `qengine` should be set to QNNPACK. We control that here - https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/Context.cpp#L122

Reply:
yeah it is another runtime dispatcher because we unified the API for fbgemm and qnnpack.

Reply:
@supriyar did you add this boilerplate code? could you talk about why did we do this? Do we ever run this code with fbgemm backend?

Reply:
@jerryzh168 see my comment above. I don't think this boilerplate is necessary because we set a default backend (for mobile it should be qnnpack) in Context.cpp.